src/commands/info/leaderboard.ts

```diff
@@ -2,38 +2,53 @@ import { SageUser } from '@lib/types/SageUser';
 import { Leaderboard } from '@root/src/lib/enums';
 import { Command } from '@lib/types/Command';
 import { createCanvas, CanvasRenderingContext2D, loadImage } from 'canvas';
-import { MessageEmbed, Message } from 'discord.js';
+import { MessageEmbed, ApplicationCommandOptionData, CommandInteraction } from 'discord.js';
 
 export default class extends Command {
 
 	description = 'Gives the top 10 users in the guild';
 	usage = '[page number]';
 	extendedHelp = 'Enter a page number to look further down the leaderboard';
 	runInDM = false;
-	aliases = ['rank', 'leader'];
 
-	async run(msg: Message, [page]: [number]): Promise<Message> {
-		msg.guild.members.fetch();
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'pagenumber',
+			description: 'leaderboard page to view',
+			type: 'NUMBER',
+			required: false
+		}
+	]
+
+	async tempRun(interaction: CommandInteraction): Promise<void> {
+		interaction.guild.members.fetch();
 
 		// eslint-disable-next-line no-extra-parens
-		const users: Array<SageUser> = (await msg.client.mongo.collection('users').find().toArray() as Array<SageUser>)
-			.filter(user => msg.guild.members.cache.has(user.discordId))
+		const users: Array<SageUser> = (await interaction.client.mongo.collection('users').find().toArray() as Array<SageUser>)
+			.filter(user => interaction.guild.members.cache.has(user.discordId))
 			.sort((ua, ub) => ua.level - ub.level !== 0 ? ua.level > ub.level ? -1 : 1 : ua.curExp < ub.curExp ? -1 : 1); // filter on level first, then remaining xp
 
+		let page = 1;
+
+		if (interaction.options.getNumber('pagenumber')) {
+			page = interaction.options.getNumber('pagenumber');
+		}
```

Comment:
The `??` operator can greatly condense this. [Docs for reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator).
```suggestion
		const page = interaction.options.getNumber('pagenumber') ?? 1;
```